remoter-server/src/dataIntegration.r

```diff
@@ -0,0 +1,156 @@
+    if (as.logical(toupper(config$enabled))){
+        # So far we only support Seurat V3
+        scdata.integrated <- run_dataIntegration(scdata, config)
+            # Compute explained variance for the plot2
+        eigValues = (scdata.integrated@reductions$pca@stdev)^2  ## EigenValues
+        varExplained = eigValues / sum(eigValues)
+        # As a short solution, we are going to store an intermediate slot for the numPCs, since this parameter is required when performing
+        # the computeEmdedding. The main reason to do not have in the config.configureEmbedding is that this parameter does not change in the configureEmbedding step.
+        scdata.integrated@misc[["numPCs"]] <- config$dimensionalityReduction$numPCs
+    }
+    else
+        scdata.integrated <- scdata
+
+    scdata.integrated <- colorObject(scdata.integrated)
+    cells_order <- rownames(scdata.integrated@meta.data)
+    plot1_data <- unname(purrr::map2(scdata.integrated@reductions$umap@cell.embeddings[, 1],scdata.integrated@reductions$umap@cell.embeddings[, 2],function(x,y){c("x"=x,"y"=y)}))
+    
+    #Adding color and sample id
+    plot1_data <- purrr::map2(plot1_data,
+        unname(scdata.integrated@meta.data[cells_order, "type"]),
+        function(x,y){append(x,list("sample"=y))}
+    )
+    plot1_data <- purrr::map2(plot1_data,
+        unname(scdata.integrated@meta.data[cells_order, "color_samples"]),
+        function(x,y){append(x,list("col"=y))}
+    )
+
+    plot2_data <- unname(purrr::map2(1:50,varExplained,function(x,y){c("PC"=x,"percentVariance"=y)}))
+
+    plots <- list()
+    plots[generate_plotuuid("", task_name, 0)] <- list(plot1_data)
+    plots[generate_plotuuid("", task_name, 1)] <- list(plot2_data)    
+
+    # For now config is not updated, since there is not new changes
+    # config <- ...
+
+    # the result object will have to conform to this format: {data, config, plotData : {plot1, plot2}}
+    result <- list(
+        data = scdata.integrated,
+        config = config,
+        plotData = plots
+    )
+
+    return(result)
+
+}
+
+# This function covers
+#   - Integrate the data using the variable "type" (in case of data integration method is selected) and normalize using LogNormalize method.
+#   - Compute PCA analysis
+#   - To visualize the results of the batch effect, an UMAP with default setting has been made. 
+# Seurat V3 pipeline (see for other methods: https://satijalab.org/seurat/archive/v3.0/integration.html)
+run_dataIntegration <- function(scdata, config){
+    
+    method <- config$dataIntegration$method
+    nfeatures <- config$dataIntegration$methodSettings[[method]]$numGenes
+    normalization <- config$dataIntegration$methodSettings[[method]]$normalisation
+    
+    #Caps lock independent just in case anything goes wrong in dynamo or wherever. 
+    if (toupper(normalization)=="LOGNORMALIZE"  | toupper(normalization)=="LOGNORMALISE") normalization<-"LogNormalize"
```

Comment:
I thought this was spelled `logNormalise` with an **s** on the UI. Did that get changed?